src/messages.rs

```diff
@@ -349,6 +352,50 @@ impl ToField for OutgoingMessages {
     }
 }
 
+fn encode_packet(message: &str) -> String {
+    let data = message.as_bytes();
+
+    let mut packet: Vec<u8> = Vec::with_capacity(data.len() + 4);
+
+    packet.write_u32::<BigEndian>(data.len() as u32).unwrap();
+    packet.write_all(data).unwrap();
+
+    std::str::from_utf8(&packet).unwrap().into()
+}
+
+#[derive(Default, Debug, Clone)]
+pub(crate) struct RequestHandshake {
+    min_version: i32,
+    max_version: i32,
+}
+impl RequestHandshake {
```

Comment:
I've created a different request type for the RequestHandshake so we can create handshake bytestrings to test with more easily